package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,414 @@
+        A matrix with each term divided by a local scale parameter
+
+    Methods
+    -------
+    determineEpsilon()
+        Determine local scale parameters using a chosen algorithm
+    """
+    def __init__(self, DistanceMatrix, **kwargs):
+        self._DistanceMatrix = DistMatrix
+
+    def determine_epsilon(self):
+        pass
+
+
+class EpsilonConstant(Epsilon):
+    """Premade function for the determination of epsilon based on providing
+    an arbitrary constant"""
+
+    def __init__(self, DistanceMatrix, epsilon):
+        """
+        Parameters
+        ----------
+        epsilon : int
+            The value of epsilon to be used as a local scale parameter
+        """
+        self.scaledMatrix = DistanceMatrix.dist_matrix
+        self._epsilon = epsilon
+
+    def determine_epsilon(self):
+        self.scaledMatrix /= self._epsilon
+        return
+
+
+class DiffusionMap(AnalysisBase):
+    """Non-linear dimension reduction method
+
+    Dimension reduction with diffusion mapping of selected structures in a
+    trajectory.
+
+    Attributes
+    ----------
+    eigenvalues: array
+        Eigenvalues of the diffusion map
+    eigenvectors: array
+        Eigenvectors of the diffusion map
+    fit : array
+        After calling `transform(num_eigenvectors)` the diffusion map embedding
+        into the lower dimensional diffusion space will exist here.
+
+    Methods
+    -------
+    decompose_kernel()
+        Constructs an anisotropic diffusion kernel and performs eigenvalue
+        decomposition on it.
+
+    spectral_gap()
+        Retrieve a guess for the set of eigenvectors reflecting the intrinsic
+        dimensionality of the molecular system.
+
+    transform(num_eigenvectors)
+        Perform an embedding of a frame into the eigenvectors representing
+        the collective coordinates.
+    """
+
+    def __init__(self, DistanceMatrix, epsilon, weights=None, timescale=1):
+        """
+        Parameters
+        -------------
+        DistanceMatrix : DistanceMatrix object
+            Distance matrix to be made into a diffusion kernel and perform
+            an eigenvector decomposition on.
+        epsilon : Epsilon object
+            Specifies the method used for the choice of scale parameter in the
+            diffusion map. More information in [1], [2] and [3].
+        weights: list, optional
+            The list has to have the same length as the trajectory.
+            With 'None' the weight of each frame of the trajectory will be the
+            same.
+        timescale: int, optional
+            The number of steps in the random walk, large t reflects global
+            structure whereas small t indicates local structure.
+        """
+        self.DistanceMatrix = DistanceMatrix
+        self._nframes = DistanceMatrix.nframes
+        self._t = timescale
+        self._epsilon = epsilon
+
+        if weights is None:
+            # weights do not apply to metric but density of data
```

Comment:
This is something you can read about in page 14 of the Lafon Paper or my diffusion maps blog post. The weight quantity is supposed to reflect how crowded points are on the free energy surface in the case of the adk simulation.